### PR TITLE
Add ability to run sidecars with consul

### PR DESCRIPTION
**Pull request by shahbazn**
Problem: This chart does not allow running sidecar containers with consul, that might me needed for some scenarios e.g. running prometheus exporter.

Solution: Add configurable sidecar containers to the chart

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -224,6 +224,10 @@ client:
     # https_proxy: http://localhost:3128,
     # no_proxy: internal.domain.com
 
+  # extraContainer is a list of side-car containers that you might need to run with your
```
Comment (lkysow) [inline values.yaml]:
```suggestion
  # extraContainers is a list of sidecar containers. Specified as a raw YAML string.
```

File: values.yaml [hunk 2]
```diff
@@ -224,6 +224,10 @@ client:
...
+  # vault instance e.g. prometheus exporter
```
Comment (lkysow) [inline values.yaml]: > vault

Someone's been copy/pasting 😆

Comment (lkysow) [inline values.yaml]:
```suggestion
```

File: values.yaml [hunk 3]
```diff
@@ -224,6 +224,10 @@ client:
...
+  extraContainers: |
```
Comment (lkysow) [inline values.yaml]:
```suggestion
  extraContainers: null
```

File: values.yaml (hunk 2 above)
Comment (shahbazn) [reply values.yaml]: hahah good catch cp source (https://github.com/hashicorp/vault-helm/pull/87/files)

**Comment by shahbazn** [comment]
Hi @lkysow I have addressed your comments and added unit tests as well.

**Comment by shahbazn** [comment]
Hi @lkysow the PR is ready for review again :)

**Comment by glothriel** [comment]
Is there any specific reason this is stuck on review? This is great feature.

**Comment by t-eckert** [comment]
Thank you for your pull request! The contents of this repository have been merged into [`hashicorp/consul-k8s`](https://github.com/hashicorp/consul-k8s) under the `charts/consul` directory. Please follow the instructions [here](https://github.com/hashicorp/consul-helm/issues/1051) to migrate your pull request to the `consul-k8s` repository.
This PR will now be closed and the `consul-helm` repository archived to avoid confusion between the old and new repositories.

**Pull request closed without merging**